service/src/main/kotlin/fi/espoo/evaka/shared/auth/JwtTokens.kt

```diff
@@ -28,37 +28,23 @@ fun DecodedJWT.toAuthenticatedUser(): AuthenticatedUser? = this.subject?.let { s
         .map { enumValueOf<UserRole>(it.removePrefix("ROLE_")) }
         .toSet()
     return when (type) {
-        "citizen" -> AuthenticatedUser.Citizen(id)
-        "citizen_weak" -> AuthenticatedUser.WeakCitizen(id)
-        "employee" -> AuthenticatedUser.Employee(id, roles)
-        "mobile" -> AuthenticatedUser.MobileDevice(id)
-        "system" -> AuthenticatedUser.SystemInternalUser
-        else -> {
-            // support legacy format temporarily
-            when {
-                id == AuthenticatedUser.SystemInternalUser.id -> AuthenticatedUser.SystemInternalUser
-                roles == setOf(UserRole.END_USER) -> AuthenticatedUser.Citizen(id)
-                roles == setOf(UserRole.MOBILE) -> AuthenticatedUser.MobileDevice(id)
-                else ->
-                    AuthenticatedUser.Employee(id, roles)
-            }
-        }
+        AuthenticatedUserType.citizen -> AuthenticatedUser.Citizen(id)
+        AuthenticatedUserType.citizen_weak -> AuthenticatedUser.WeakCitizen(id)
+        AuthenticatedUserType.employee -> AuthenticatedUser.Employee(id, roles)
+        AuthenticatedUserType.mobile -> AuthenticatedUser.MobileDevice(id)
+        AuthenticatedUserType.system -> AuthenticatedUser.SystemInternalUser
+        null -> null
     }
 }
 
 fun AuthenticatedUser.applyToJwt(jwt: JWTCreator.Builder): JWTCreator.Builder = jwt
     .withSubject(id.toString())
-    .withClaim(
-        "evaka_type",
-        when (this) {
-            is AuthenticatedUser.SystemInternalUser -> "system"
-            is AuthenticatedUser.MobileDevice -> "mobile"
-            is AuthenticatedUser.Employee -> "employee"
-            is AuthenticatedUser.Citizen -> "citizen"
-            is AuthenticatedUser.WeakCitizen -> "citizen_weak"
+    .withClaim("evaka_type", this.type.toString())
+    .also {
+        if (this is AuthenticatedUser.Employee) {
```

Comment:
Why don't other users have their roles attached?

Reply:
Because none of the other user types have configurable roles and the "roles" are hard-coded based on the user type. The only reason those special roles even exist is legacy reasons where we needed to have one unified big role array for permission checks.
Once we implement action-based permissions, `END_USER`/`CITIZEN_WEAK`/`MOBILE_DEVICE` roles can be removed completely because they no longer have a purpose.